shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/ingest/dumper/AbstractInventoryDumper.java

```diff
@@ -167,12 +167,11 @@ private Optional<Object> dump0(final Connection conn, final String sql, final in
                     record.setType(IngestDataChangeType.INSERT);
                     record.setTableName(logicTableName);
                     for (int i = 1; i <= metaData.getColumnCount(); i++) {
-                        boolean isUniqueKey = tableMetaData.isUniqueKey(i - 1);
                         Object value = readValue(resultSet, i);
-                        if (isUniqueKey) {
+                        if (tableMetaData.isPrimaryKey(i - 1)) {
                             maxUniqueKeyValue = value;
                         }
-                        record.addColumn(new Column(metaData.getColumnName(i), value, true, isUniqueKey));
+                        record.addColumn(new Column(metaData.getColumnName(i), value, true, tableMetaData.isUniqueKey(i - 1)));
                     }
```

Comment:
Rollback might be not enough, since we need to support unique key (without primary key) table migration